syntax = "proto3";

package ray.streaming.proto;

option java_package = "org.ray.streaming.runtime.generated";

enum OperatorType {
  UNKNOWN = 0;
  TRANSFORM = 1;
  SOURCE = 2;
  SINK = 3;
}

enum FlowControlType {
  UNKNOWN_FLOW_CONTROL_TYPE = 0;
  UnconsumedSeqFlowControl = 1;
  NoFlowControl = 2;
}

// all string in this message is ASCII string
message StreamingConfig {
  string job_name = 1;
  string task_job_id = 2;
  string worker_name = 3;
  string op_name = 4;
  OperatorType role = 5;
  uint32 ring_buffer_capacity = 6;
  uint32 empty_message_interval = 7;
  FlowControlType flow_control_type = 8;
  uint32 writer_consumed_step = 9;
  uint32 reader_consumed_step = 10;
  uint32 event_driven_flow_control_interval = 11;
}
